saleor/graphql/shop/mutations.py

```diff
@@ -397,18 +409,23 @@ def perform_mutation(cls, _root, info, **data):
             "automatically_confirm_all_new_orders",
             "automatically_fulfill_non_shippable_gift_card",
         ]
-        site = get_site_promise(info.context).get()
-        instance = site.settings
-        update_fields = []
+        update_fields = {}
         for field in FIELDS:
-            value = data["input"].get(field)
-            if value is not None:
-                setattr(instance, field, value)
-                update_fields.append(field)
+            if field in data["input"]:
+                update_fields[field] = data["input"][field]
 
         if update_fields:
-            instance.save(update_fields=update_fields)
-        return OrderSettingsUpdate(order_settings=instance)
+            channel_models.Channel.objects.update(**update_fields)
+        channel = channel_models.Channel.objects.order_by("slug").first()
```

Comment:
I'm not sure we should update only one channel in this situation. We shouldn't guess in which channel those changes should be applied. 
IMO if we want to leave this behavior as deprecated then these mutations should change those values in all channels or require a channel to update.

Reply:
But, we do update all channels here. Or I'm missing something?
I mean, the line: `channel_models.Channel.objects.update(**update_fields)` is updating every channel.

Reply:
Getting first channel is only to simplify logic in getting the value to return in mutation.
Note: The logic of getting first channel to show the `orderSettings` will be updated the same way as you are sugesting (that it should be first active channel)

Reply:
My mistake. The code looks good in this place.